`invalid-name` type alias false-positive for union variables without assignment values

### Bug description

```
# pylint: disable=missing-docstring
from typing import Union
x: Union[str, int]
```

### Configuration

_No response_

### Command used

```shell
pylint t.py
```


### Pylint output

```shell
************* Module t
t.py:3:0: C0103: Type alias name "x" doesn't conform to predefined naming style (invalid-name)
```


### Expected behavior

No output.

### Pylint version

```shell
pylint 2.17.2
astroid 2.15.2
Python 3.11.1 (main, Jan 10 2023, 15:08:01) [Clang 14.0.0 (clang-1400.0.29.202)]
```


### OS / Environment

_No response_

### Additional dependencies

Similar to #8487. This is a bug found in #8537, let me send a separate PR first.

--- a/doc/whatsnew/fragments/8540.false_positive
+++ b/doc/whatsnew/fragments/8540.false_positive
@@ -1,0 +1,4 @@
+`Union` typed variables without assignment are no longer treated as
+`TypeAlias`.
+
+Closes #8540

--- a/pylint/checkers/base/name_checker/checker.py
+++ b/pylint/checkers/base/name_checker/checker.py
@@ -602,10 +602,7 @@ class NameChecker(_BasicChecker):
                 # Union is a special case because it can be used as a type alias
                 # or as a type annotation. We only want to check the former.
                 assert node is not None
-                return not (
-                    isinstance(node.parent, nodes.AnnAssign)
-                    and node.parent.value is not None
-                )
+                return not isinstance(node.parent, nodes.AnnAssign)
         elif isinstance(inferred, nodes.FunctionDef):
             if inferred.qname() == "typing.TypeAlias":
                 return True

--- a/tests/functional/t/typealias_naming_style_default.py
+++ b/tests/functional/t/typealias_naming_style_default.py
@@ -26,5 +26,8 @@ _1BadName = Union[int, str]  # [invalid-name]
 ANOTHERBADNAME = Union[int, str]  # [invalid-name]
 
 # Regression tests
-# This is not a TypeAlias, and thus shouldn't flag the message
+# They are not TypeAlias, and thus shouldn't flag the message
 x: Union[str, int] = 42
+y: Union[str, int]
+# But the following, using a good TypeAlias name, is:
+GoodTypeAliasToUnion: TypeAlias = Union[str, int]

--- a/tests/functional/t/typealias_naming_style_rgx.py
+++ b/tests/functional/t/typealias_naming_style_rgx.py
@@ -3,8 +3,8 @@ from typing import TypeAlias, Union
 
 # Valid
 TypeAliasShouldBeLikeThis: TypeAlias = int
-_TypeAliasShouldBeLikeThis: Union[str, int]
+_TypeAliasShouldBeLikeThis = Union[str, int]
 
 # Invalid
 TypeAliasShouldntBeLikeThis: TypeAlias = int  # [invalid-name]
-_TypeAliasShouldntBeLikeThis: Union[str, int]  # [invalid-name]
+_TypeAliasShouldntBeLikeThis = Union[str, int]  # [invalid-name]

